plugins/database/src/db_plugin.cpp

```diff
@@ -2992,6 +2994,19 @@ irods::error db_reg_replica_op(
     cVal[IX_DATA_PATH]       = _dst_data_obj_info->filePath;
     cVal[IX_DATA_MODE]       = _dst_data_obj_info->dataMode;
 
+    // The caller has requested that the replica be registered as intermediate. This
+    // means that the replica will be written or changed at a future time. Otherwise,
+    // the replica will take the replica status of the source replica.
+    char* intermediate_replica_status = "2";
```

Comment:
this isn't defined somewhere else already as '2'?

Reply:
What I normally do is a `std::to_string` on the `INTERMEDIATE_REPLICA` enumerated value, but that won't play well here as `cVal` is an array of pointer to non-const `char`s. I could cast away the const or perform a `std::snprintf` to a pre-existing char array. Casting away the const seems pretty safe given it's coming from a temporary anyway.